src/mscorlib/shared/System/Threading/Tasks/IValueTaskSource.cs

```diff
@@ -0,0 +1,77 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Threading.Tasks
+{
+    /// <summary>
+    /// Flags passed from <see cref="ValueTask"/> and <see cref="ValueTask{TResult}"/> to
+    /// <see cref="IValueTaskSource.OnCompleted"/> and <see cref="IValueTaskSource{TResult}.OnCompleted"/>
+    /// to control behavior.
+    /// </summary>
+    [Flags]
+    public enum ValueTaskSourceOnCompletedFlags
+    {
+        /// <summary>
+        /// No requirements are placed on how the continuation is invoked.
+        /// </summary>
+        None,
+        /// <summary>
+        /// Set if OnCompleted should capture the current scheduling context (e.g. SynchronizationContext)
+        /// and use it when queueing the continuation for execution.  If this is not set, the implementation
+        /// may choose to execute the continuation in an arbitrary location.
+        /// </summary>
+        UseSchedulingContext = 0x1,
+        /// <summary>
+        /// Set if OnCompleted should capture the current <see cref="ExecutionContext"/> and use it to
+        /// <see cref="ExecutionContext.Run"/> the continuation.
+        /// </summary>
+        FlowExecutionContext = 0x2,
+    }
+
+    /// <summary>Indicates the status of an <see cref="IValueTaskSource"/> or <see cref="IValueTaskSource{TResult}"/>.</summary>
+    public enum ValueTaskSourceStatus
+    {
+        /// <summary>The operation has not yet completed.</summary>
+        Pending = 0,
+        /// <summary>The operation completed successfully.</summary>
+        Succeeded = 1,
+        /// <summary>The operation completed with an error.</summary>
+        Faulted = 2,
+        /// <summary>The operation completed due to cancellation.</summary>
+        Canceled = 3
+    }
+
+    /// <summary>Represents an object that can be wrapped by a <see cref="ValueTask"/>.</summary>
+    public interface IValueTaskSource
+    {
+        /// <summary>Gets the status of the current operation.</summary>
+        ValueTaskSourceStatus Status { get; }
```

Comment:
Are we not doing the cookies?

Reply:
See https://github.com/dotnet/coreclr/pull/16618#issuecomment-369291758.

Reply:
For some reason when I click on the link it does not take me anywhere.

Reply:
> For some reason when I click on the link it does not take me anywhere.

Hmm, ok, well it's a top-level comment earlier in this PR thread that outlines a few remaining action items to follow-up one and determine whether to address or ignore, one of which was cookies, another of which was namespace.

Reply:
Don't feel strongly about whether they are added; but if it were something like `CorrelationToken` would be better than `Cookie`

Reply:
> Don't feel strongly about whether they are added; but if it were something like CorrelationToken would be better than Cookie

You mean just in terms of name? _If_ we were to do something (I'm still investigating any perf ramifications), I was thinking it would just be something like `short token`.